### ISSUE TITLE
Strange warning at return SupportsInt

### ISSUE THREAD

**Issue by jolaf** [agronholm/typeguard#83]
The following code:
```python3
from typing import SupportsInt

from typeguard import TypeChecker

def f() -> SupportsInt:
    return 5

with TypeChecker(('__main__',)):
    f()
```
produces the following output:
```python3
$ python3 Test.py 
/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py:932: TypeWarning: [MainThread] return from __main__.f() at Test.py:6: Protocols cannot be used with isinstance().
  warn(TypeWarning(memo, event, frame, exc))
```

**Comment by agronholm** [comment]
If I cannot use `isinstance()` with these, how can I match them with the code path I need to execute?

**Comment by jolaf** [comment]
Well, generally it's a tough questions, as protocols are "supported functionality definitions", not superclasses.

Therefore I see the following possible approaches:

1. For Protocols, instead of `isinstance(obj, cls)` use `issubclass(type(obj), cls)`, it works.
This is not full support, but at least it would cover the most common cases.

2. Have internal table of known protocols that would contain, for each protocol, a function that would check if an object has the necessary interfaces. For example, for `SupportsInt` protocol, it's a `def __int__(self) -> int` method. See https://docs.python.org/3/library/typing.html?highlight=supportsint#typing.SupportsInt for details.

3. Have internal table of known protocols that would contain, for each protocol, a function that would somehow check that an object actually conforms to that protocol. For example, for `SupportsInt` it may be calling `int(obj)` and making sure no exceptions gets thrown.

4. Check code of, say, `mypy` that handles this somehow, while not in runtime.

**Comment by jolaf** [comment]
See also:
https://github.com/python/typing/issues/297
https://stackoverflow.com/questions/34844514

**Comment by jolaf** [comment]
Crashes with import hook as well.

`TypeGuardTest.py`:
```python3
from typeguard.importhook import install_import_hook

install_import_hook('Test')

import Test
```
`Test.py`:
```python3
from typing import SupportsInt

def f() -> SupportsInt:
    return 5

f()
```
```python3
$ python3 TypeGuardTest.py 
Traceback (most recent call last):
  File "TypeGuardTest.py", line 5, in <module>
    import Test
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "/usr/local/lib/python3.6/dist-packages/typeguard/importhook.py", line 80, in exec_module
    return super().exec_module(module)
  File "/media/sf_d/WWPass/git/Test.py", line 6, in <module>
    f()
  File "/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py", line 735, in wrapper
    check_return_type(retval, memo)
  File "/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py", line 581, in check_return_type
    raise exc from None
  File "/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py", line 579, in check_return_type
    check_type('the return value', retval, memo.type_hints['return'], memo)
  File "/usr/local/lib/python3.6/dist-packages/typeguard/__init__.py", line 541, in check_type
    if not isinstance(value, expected_type):
  File "/usr/lib/python3.6/typing.py", line 1640, in __instancecheck__
    raise TypeError("Protocols cannot be used with isinstance().")
TypeError: Protocols cannot be used with isinstance().
```

**Comment by jolaf** [comment]
As of v2.6.1 this one (together with #85) seems like the last critical problem preventing me from using the import hook mode full-scale for my software. So I'd be very grateful if the priority could be risen for this one. Thanks!

**Comment by agronholm** [comment]
I'll try to do something about this next weekend. I've been spending quite a few of them fixing up typeguard. As for #85, I'm planning to make a proper version of the warnings support feature in v3.0. However, I am keeping up the `warnings` branch with master which should suffice while you wait.

**Comment by jolaf** [comment]
Thanks!!

**Comment by jolaf** [comment]
Is there a plan for a release including this fix? Thanks!

**Comment by agronholm** [comment]
Released in v2.7.0.

**Comment by jolaf** [comment]
Thanks, it was very helpful!
